src/kvstore/NebulaStore.cpp

```diff
@@ -512,6 +524,36 @@ ResultCode NebulaStore::flush(GraphSpaceID spaceId) {
     return ResultCode::SUCCEEDED;
 }
 
+ResultCode NebulaStore::createSnapshot(GraphSpaceID spaceId) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        return error(spaceRet);
+    }
+    auto space = nebula::value(spaceRet);
+    for (auto& engine : space->engines_) {
+        auto code = engine->createSnapshot();
+        if (code != ResultCode::SUCCEEDED) {
+            return code;
+        }
+    }
+    return ResultCode::SUCCEEDED;
+}
+
+ResultCode NebulaStore::deleteSnapshot(GraphSpaceID spaceId) {
+    auto spaceRet = space(spaceId);
+    if (!ok(spaceRet)) {
+        return error(spaceRet);
+    }
+    auto space = nebula::value(spaceRet);
+    for (auto& engine : space->engines_) {
+        auto code = engine->deleteSnapshot();
+        if (code != ResultCode::SUCCEEDED) {
+            return code;
```

Comment:
return directly may cause some engine's snapshot has been deleted, but some not be deleted?

Reply:
Keen eye 👍 .  deleteSnapshot() will always return SUCCEEDED. There is no leakage of snapshot. but the code still needs improvement.